util/plot_utils.py

```diff
@@ -43,6 +44,14 @@ def plot_logs(logs, fields=('class_error', 'loss_bbox_unscaled', 'mAP'), ewm_col
             continue
         raise ValueError(f"{func_name} - invalid directory in logs argument:\n{dir}")
 
+    # check if first log file exists...it will not be present until after epoch 1.
+    fn = Path(logs[0] / log_name)
+
+    if not fn.exists():
+        print(f"-> missing {log_name} in first directory.  Have you gotten to Epoch 1 in training?")
+        print(f"--> file path: {fn}")
+        return
```

Comment:
In theory, the user can specify many log files here, so this should maybe be something like
```python
for log_file in logs:
    fn = Path(log_file / log_name)
    # rest of the check per log
```

Reply:
very valid point @fmassa - I prefer to add this full check now, so we can check in a full and robust PR upfront.  
 (when I was at MSFT I often mentioned the phrase 'there's never time to do it right, but always time to do it over' for decisions like this, so prefer to do it right up front. )

Reply:
we're already looping through to check each dir in the code, so the full log_name in each dir check integrates properly there. 
 
I've thus rewritten that main loop into a single quality check loop where we verify:
1 - each item in log list is a Path object
2 - each directory exists
3 - that each directory holds a valid {log_name} file

![improved-missing-log-check](https://user-images.githubusercontent.com/46302957/89812931-bcf16200-daf5-11ea-8bd7-6cb92c96b532.PNG)

Tested valid sets of directories/log files and also with missing log files.  
Error cases look much nicer now vs a spew of code with 'object missing' at the end for a missing log file scenario.